src/main/java/beam/physsim/jdeqsim/cacc/sim/Road.java

```diff
@@ -191,23 +195,23 @@ public void procsessFilledRoad(org.matsim.core.mobsim.jdeqsim.Vehicle vehicle, d
             nextStuckTime=simTime + config.getSqueezeTime();
         }
 
-        if (!Road.getRoad(vehicle.getCurrentLinkId()).latestTimeToLeaveRoad.containsKey(vehicle)){
-            Road.getRoad(vehicle.getCurrentLinkId()).latestTimeToLeaveRoad.put(vehicle,simTime + link.getLength()/minimumRoadSpeedInMetersPerSecond);
+        if (!getRoad(vehicle.getCurrentLinkId()).latestTimeToLeaveRoad.containsKey(vehicle)){
+            getRoad(vehicle.getCurrentLinkId()).latestTimeToLeaveRoad.put(vehicle,simTime + link.getLength()/minimumRoadSpeedInMetersPerSecond);
         }
 
         double minTimeForNextDeadlockPreventionMessageTime=0;
 
         if (getDeadlockPreventionMessages().size() > 0) minTimeForNextDeadlockPreventionMessageTime=
                 getDeadlockPreventionMessages().getLast().getMessageArrivalTime()+INCREASE_TIMESTAMP; // ensures that deadlock prevention messages have increasing time stamps - this is assumped by original implementation around this
 
-        double timeToLeaveRoad=Math.max(Math.min(Road.getRoad(vehicle.getCurrentLinkId()).latestTimeToLeaveRoad.get(vehicle),nextStuckTime),minTimeForNextDeadlockPreventionMessageTime);
+        double timeToLeaveRoad=Math.max(Math.min(getRoad(vehicle.getCurrentLinkId()).latestTimeToLeaveRoad.get(vehicle),nextStuckTime),minTimeForNextDeadlockPreventionMessageTime);
 
         getDeadlockPreventionMessages().add(vehicle.scheduleDeadlockPreventionMessage(timeToLeaveRoad, this));
 
         assert (getInterestedInEnteringRoad().size()== getDeadlockPreventionMessages().size()) :getInterestedInEnteringRoad().size() + " - " + getDeadlockPreventionMessages().size();
     }
 
-    public static Road getRoad(Id<Link> linkId) {
-        return (Road) getAllRoads().get(linkId);
+    public Road getRoad(Id<Link> linkId) {
+        return (Road) allRoads.get(linkId);
```

Comment:
Again - maybe this is managed more in matsim now, but I would think that we would want to sync with the underlying mechanism and keep using the `getAllRoads`

Reply:
Another solution would be running beam and tests using different classloaders. I have not researched this possibility yet.

Reply:
After looking at the other PR I think this seems to be more semantic than anything else....although the underlying allRoads issue is up for grabs there.